includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1587,48 +1398,18 @@ public function generate_account_statement_descriptor_html( $key, $data ) {
 	 *
 	 * @param  string $key           Option key.
 	 * @param  mixed  $empty_value   Value when empty.
-	 * @return string|array|int|bool The value specified for the option or a default value for the option.
+	 * @return string|array|int|bool|null The value specified for the option or a default value for the option.
 	 */
 	public function get_option( $key, $empty_value = null ) {
-		switch ( $key ) {
-			case 'enabled':
-				return parent::get_option( static::METHOD_ENABLED_KEY, $empty_value );
-			case 'account_statement_descriptor':
-				return $this->get_account_statement_descriptor();
-			case 'account_business_name':
-				return $this->get_account_business_name();
-			case 'account_business_url':
-				return $this->get_account_business_url();
-			case 'account_business_support_address':
-				return $this->get_account_business_support_address();
-			case 'account_business_support_email':
-				return $this->get_account_business_support_email();
-			case 'account_business_support_phone':
-				return $this->get_account_business_support_phone();
-			case 'account_branding_logo':
-				return $this->get_account_branding_logo();
-			case 'account_branding_icon':
-				return $this->get_account_branding_icon();
-			case 'account_branding_primary_color':
-				return $this->get_account_branding_primary_color();
-			case 'account_branding_secondary_color':
-				return $this->get_account_branding_secondary_color();
-			case 'deposit_schedule_interval':
-				return $this->get_deposit_schedule_interval();
-			case 'deposit_schedule_weekly_anchor':
-				return $this->get_deposit_schedule_weekly_anchor();
-			case 'deposit_schedule_monthly_anchor':
-				return $this->get_deposit_schedule_monthly_anchor();
-			case 'deposit_delay_days':
-				return $this->get_deposit_delay_days();
-			case 'deposit_status':
-				return $this->get_deposit_status();
-			case 'deposit_completed_waiting_period':
-				return $this->get_deposit_completed_waiting_period();
+		if ( in_array( $key, WC_Payments_Gateway_WCPay_Settings::OPTIONS, true ) ) {
```

Comment:
Do you think a more drastic approach where **all** options are accessed through `WC_Payments_Gateway_WCPay_Settings` might make sense? Here's what I mean:

1. `WC_Payments_Gateway_WCPay_Settings` is constructed with a dependency on the gateway in order to be able to access `$this->gateway->get_option()` for settings/as a fallback.
2. All usages of `WC_Payments()->get_gateway()->get_option()` are replaced with proper getters within the settings class, ex. `WC_Payments()->get_settings()->is_enabled()`.
3. This method (`get_option` here) is completely removed.

I know this will involve a lot of file changes (ex. there are 36 changed files in https://github.com/Automattic/woocommerce-payments/pull/4986 just for `is_in_test_mode()` and `is_in_dev_mode()`), but I see a few benefits:

- ~It would not matter~ The gateway would not need to know where the option comes from (cached account data vs gateway settings).
- All options will be clearly accessible from a single place.
- We'd be able to utilize proper types/casting.

Expanding a bit further, we could also use the field types from `WC_Payments_Gateway_WCPay_Settings::get_form_fields()` and parse the value, ending the annoying `'yes' === $gateway->get_option( 'something' )`, which also premises bugs (imagine assuming it's a boolean, then even 'no' will be true). This can be done either through magic methods (`__get` for `$settings->enabled`) and PHPDoc `@property` hints (see [this example](https://github.com/Automattic/woocommerce-payments/pull/4986/files#diff-1f0e10aa8fe69406fae8d3d26368dd8b46e67c541485b5a84669086bb0150f2fR16)), or explicit methods like this:

```php
/**
 * @property-read bool $enabled Indicates if the gateway is enabled.
 */
class {
	private function get_option( $name ) {
		$value = $this->gateway->get_option( $name );

		// Of course, this formatting would change.
		$type = $this->get_form_fields()[ $name ]['type'];
		switch ( $type ) {
			case 'checkbox':
				$value = ( 'yes' === $value );
				break;
		}

		return $value;
	}

	public function is_enabled() : bool {
		return $this->get_option( 'enabled' );
	}

	public function __get( $name ) {
		if ( isset( $this->get_form_fields()[ $name ] ) ) {
			return $this->get_option( $name );
		} else {
			// throw exception
		}
	}
}
```

This example combines both approaches, but we should stick to only one of them.

Reply:
`WC_Payment_Gateway_WCPay` already depend on `Settings` if we make `Settings` depend on `WC_Payment_Gateway_WCPay` then we would have a circular reference. 

This could be avoided by using `WC_Payments::get_gateway_settings` but that's not the most elegant way to do this. Though we already have code like that in other places. So I am not totally against it but not a big fan either. 

What do you think?